File: lesson-07/js/resize.js
var array = Array(0);
var index = 0; // 3

const append = (item) => {
    var n = array.length;
    if (index >= n) {
        const temp = Array(n === 0 ? 1 : 2 * n);
        for (let i = 0; i < index; i++) {
            temp[i] = array[i];
        }
        array = temp;
    }
    array[index++] = item;
}

const main = () => {
    const k = 5;
    const n = Math.pow(2, k) + 1;

    for (let i = 0; i <= n; i++) {
        console.log(`i: ${i}  index: ${index}  length: ${array.length}`);
        append(i);
    }
}

main();

File: lesson-07/java/Resize.java
public class Resize {
    static int[] array = new int[0];
    static int index = 0;

    static void append(int item) {
        int n = array.length;
        if (index >= n) {
            int[] temp = new int[n == 0 ? 1 : 2 * n];
            for (int i = 0; i < n; i++) {
                temp[i] = array[i];
            }
            array = temp;
        }
        array[index++] = item;
    }

    public static void main(String[] args) {
        int k = 5;
        double n = Math.pow(2, k) + 1;

        for (int i = 0; i <= n; i++) {
            System.out.printf("i: %d  index: %d  length: %d\n", i, index, array.length);
            append(i);
        }
    }
}
